File: EIDSS v6/android.java/workspace/EIDSS/src/com/bv/eidss/web/VetCaseInfoOut.java
package com.bv.eidss.web;

import java.text.ParseException;

import org.json.JSONException;
import org.json.JSONObject;

public class VetCaseInfoOut {

    public Long id;
    public String offlineCaseID;
    public String caseID;
    public String farmCode;
    public Long herd;
    public String reportedByOrganization;
    public String reportedByPerson;

    public String lastErrorDescription;

    public boolean isCreated;
    public boolean isUpdated;
    public boolean isFailed;
    
    public VetCaseInfoOut(JSONObject json) throws JSONException, ParseException{
        id = json.getLong("id");
    	offlineCaseID = json.getString("offlineCaseID");
    	caseID = json.getString("caseID");
    	farmCode = json.getString("farmCode");
    	herd = json.getLong("herd");
    	reportedByOrganization = json.getString("reportedByOrganization");
    	reportedByPerson = json.getString("reportedByPerson");
    	lastErrorDescription = json.getString("lastErrorDescription");
    	isCreated = json.getBoolean("isCreated");
    	isUpdated = json.getBoolean("isUpdated");
    	isFailed = json.getBoolean("isFailed");
    }
}

File: EIDSS v6/eidss.smartphone.web/Models/VetCaseInfoOut.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace eidss.smartphone.web.Models
{
    public class VetCaseInfoOut
    {
        public long id { get; set; }
        public string offlineCaseID { get; set; }
        public string caseID { get; set; }
        public string farmCode { get; set; }
        public long herd { get; set; }
        public string reportedByOrganization { get; set; }
        public string reportedByPerson { get; set; }

        public string lastErrorDescription { get; set; }

        public bool isCreated { get; set; }
        public bool isUpdated { get; set; }
        public bool isFailed { get; set; }

        public VetCaseInfoOut(VetCaseInfoIn vcin)
        {
            id = vcin.id;
            offlineCaseID = vcin.offlineCaseID;
            caseID = vcin.caseID;
            herd = vcin.species.Count > 0 ? vcin.species.First().herd : 0;
            lastErrorDescription = "";
        }
    }
}